lib/active_merchant/billing/gateways/blue_snap.rb

```diff
@@ -102,10 +102,13 @@ def capture(money, authorization, options = {})
       end
 
       def refund(money, authorization, options = {})
-        commit(:refund, options, :put) do |doc|
-          add_authorization(doc, authorization)
-          add_amount(doc, money, options)
-          add_order(doc, options)
+        options[:endpoint] = options[:merchant_transaction_id] ? "/refund/merchant/#{options[:merchant_transaction_id]}" : "/refund/#{authorization}"
+        commit(:refund, options, :post) do |doc|
+          send_when_present(doc, :reason, options)
+          send_when_present(doc, :cancel_subscription, options)
+          add_amount(doc, money, options) if money
+          send_when_present(doc, :tax_amount, options)
+          add_transaction_meta_data(doc, options)
```

Comment:
Could be nice to simplify this into:
```
commit(:refund, options, :post) do |doc|
  add_amount(doc, money, options) if money
  %i[reason cancel_subscription tax_amount].each { |field| send_when_present(doc, field, options) }
  add_transaction_meta_data(doc, options)
end
```

Reply:
Thanks, that looks much cleaner. Never thought of this 😄